overlord/snapstate/handlers.go

```diff
@@ -1942,6 +1942,46 @@ func (m *SnapManager) doPreferAliases(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// reRefreshSetup holds the necessary details to re-refresh snaps that need it
+type reRefreshSetup struct {
+	Snaps  []string `json:"snaps,omitempty"`
+	UserID int      `json:"user-id,omitempty"`
+	*Flags
+}
+
+var reRefreshUpdater = updateManyFromChange
+
+func (m *SnapManager) doReRefresh(t *state.Task, tomb *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	var re reRefreshSetup
+	if err := t.Get("rerefresh-setup", &re); err != nil {
+		return err
+	}
+
+	chg := t.Change()
+	updated, tasksets, err := reRefreshUpdater(tomb.Context(nil), st, chg.ID(), re.Snaps, re.UserID, re.Flags)
```

Comment:
we should only to refreshes that would change the epoch again, no?  from  0* to 1 or something?